pointer-masking-proposal.adoc

```diff
@@ -1,3 +1,370 @@
-# *RISC-V Pointer Masking Proposal*
+[[risc-v-pointer-masking-proposal]]
+# RISC-V Pointer Masking proposal
 
-(Placeholder)
+_Adam Zabrocki_, _Lee Campbell_, _Martin Maas_, _RISC-V TEE_ and _J Extension Task Groups_
+
+**Contributors: ** [Members of the TEE and J Extension Task Groups]
+
+[[introduction]]
+## Introduction
+
+RISC-V Pointer Masking (PM) is a feature which provides a possibility of implementing “memory tagging” by ignoring various bits (defined by mask) of the [.underline]#_effective address_# (virtual or physical) on RV128, RV64 and RV32.
```

Comment:
Which memory references are targeted with this proposal: only data ones, or also instruction fetch?

Reply:
Pointer masking applies both to data and to instruction accesses.

Reply:
@martinmaas if we simplify the scenario just to data, would it break any of the scenarios which you have or any scenario which you are aware of?

Reply:
It would make sandboxing for arbitrary assembly code significantly more tricky. Based on the discussion on the mailing list, it looks like we are heading in this direction though.